tasks/ruby.rb

```diff
@@ -21,10 +21,20 @@ def task(opts = {})
     stdout, stderr, status = Open3.capture3("#{facter_executable} -v")
 
     if stdout =~ /^[0-2]\./
-      exec("#{facter_executable} -p --json")
+      facter_args = "-p --json"
     else
-      exec("#{facter_executable} -p --json --show-legacy")
+      facter_args = "-p --json --show-legacy"
     end
+
+    _stdout, _stderr, _status = Open3.capture3("#{facter_executable} #{facter_args}")
+
+    result = JSON.parse(_stdout)
+
+    if _status.exitstatus != 0
+      result[:_error] = { msg: _stderr }
+    end
```

Comment:
```suggestion
    stdout, stderr, status = Open3.capture3("#{facter_executable} #{facter_args}")

    result = JSON.parse(stdout)

    if status.exitstatus != 0
      result[:_error] = { msg: stderr }
    end
```

`stdout`, `stderr`, and `status` are all used, so they shouldn't have underscores.